core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -285,13 +285,10 @@ private[spark] object Utils extends Logging {
    */
   def createDirectory(dir: File): Boolean = {
     try {
-      // This sporadically fails - not sure why ... !dir.exists() && !dir.mkdirs()
-      // So attempting to create and then check if directory was created or not.
-      dir.mkdirs()
-      if ( !dir.exists() || !dir.isDirectory) {
-        logError(s"Failed to create directory " + dir)
-      }
-      dir.isDirectory
+      // SPARK-35907
+      // This could throw more meaningful exception information if directory creation failed.
+      Files.createDirectories(dir.toPath)
+      true
```

Comment:
Hm..this looks like a behavior change to me. Are you sure `Files.createDirectories` would always not fail silently?

Otherwise, I'd prefer to keep:
```scala
if ( !dir.exists() || !dir.isDirectory) {
 logError(s"Failed to create directory " + dir)
}
dir.isDirectory
```

Reply:
In some extreme scenarios, Files.createDirectories would fail silently. In this case, we can submit an issue to the JDK official. ︿(￣︶￣)︿ The purpose of using this method is to provide better exception handling mechanism, but it should not be at the cost of reducing security, so I take your suggestion.Thanks a lot. (*^▽^*)